src/WebJobs.Script/Extensions/IServiceProviderExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Script;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class IServiceProviderExtensions
+    {
+        /// <summary>
+        /// Gets the specified ScriptHost level service, or null.
+        /// </summary>
+        /// <typeparam name="TService">The service to query for.</typeparam>
+        /// <param name="serviceProvider">The <see cref="IServiceProvider"/> to query.</param>
+        /// <returns>The service or null.</returns>
+        public static TService GetHostServiceOrNull<TService>(this IServiceProvider serviceProvider) where TService : class
```

Comment:
Rename this to `GetScriptHostServiceOrNull` to be a little clearer? (probably rename the Utility method as well).

Reply:
Agreed - will rename